lib/modules/manager/npm/post-update/pnpm.ts

```diff
@@ -66,16 +68,33 @@ export async function generateLockFile(
       extraEnv.NPM_AUTH = env.NPM_AUTH;
       extraEnv.NPM_EMAIL = env.NPM_EMAIL;
     }
-    const commands: string[] = [];
 
-    cmd = 'pnpm';
-    let args = 'install --recursive --lockfile-only';
+    let args = '--recursive --lockfile-only';
     if (!GlobalConfig.get('allowScripts') || config.ignoreScripts) {
       args += ' --ignore-scripts';
       args += ' --ignore-pnpmfile';
     }
-    logger.trace({ cmd, args }, 'pnpm command');
-    commands.push(`${cmd} ${args}`);
+    logger.trace({ args }, 'pnpm command options');
+
+    if (!upgrades.every((upgrade) => upgrade.isLockfileUpdate)) {
+      // This command updates the lock file based on package.json
+      commands.push(`pnpm install ${args}`);
```

Comment:
`pnpm install` is originally for installing all packages in `package.json` and alias for `pnpm add`. The primary command for installing a specific package is `pnpm add`.

https://pnpm.io/cli/install
https://pnpm.io/cli/add

```suggestion
      commands.push(`pnpm add ${args}`);
```